### ISSUE TITLE
yarn upgrade-interactive exits with error

### ISSUE THREAD

**Issue by maxlapides** [yarnpkg/yarn#4346]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Prior to v1, if you CTRL+C out of `yarn upgrade-interactive`, it would exit silently. Now, it will exit with this error:

```
Error: Promise rejected with value: undefined
    at /usr/local/Cellar/yarn/1.0.1/libexec/lib/cli.js:71593:12
    at Array.forEach (native)
    at EventEmitter.<anonymous> (/usr/local/Cellar/yarn/1.0.1/libexec/lib/cli.js:71589:24)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:191:7)
    at emit (/usr/local/Cellar/yarn/1.0.1/libexec/lib/cli.js:67713:11)
    at processEmit [as emit] (/usr/local/Cellar/yarn/1.0.1/libexec/lib/cli.js:67786:5)
```

**If the current behavior is a bug, please provide the steps to reproduce.**
1. run `yarn upgrade-interactive`
2. type `CTRL+C`

**What is the expected behavior?**
Exits without error

**Please mention your node.js, yarn and operating system version.**
node.js v6.11.1
yarn v1.0.1
macOS 10.12.6

**Comment by BYK** [comment]
@torifat any progress?

**Comment by maxlapides** [comment]
Looks like this is fixed in 1.0.2?

**Comment by BYK** [comment]
Oh, right. Forgot to link the PR. Fixed in #4434.
